pylint/message/message.py

```diff
@@ -41,33 +43,30 @@ def __new__(
         symbol: str,
         location: MessageLocationTuple,
         msg: str,
-        confidence: Optional[Confidence],
-    ) -> "Message":
+        confidence: Confidence | None,
+    ) -> Message:
         ...
 
     @overload
     def __new__(
         cls,
         msg_id: str,
         symbol: str,
-        location: Tuple[str, str, str, str, int, int],
+        location: tuple[str, str, str, str, int, int],
         msg: str,
-        confidence: Optional[Confidence],
-    ) -> "Message":
+        confidence: Confidence | None,
+    ) -> Message:
         # Remove for pylint 3.0
         ...
 
     def __new__(
         cls,
         msg_id: str,
         symbol: str,
-        location: Union[
-            Tuple[str, str, str, str, int, int],
-            MessageLocationTuple,
-        ],
+        location: (tuple[str, str, str, str, int, int] | MessageLocationTuple),
```

Comment:
The brackets could have been removed here though.
```diff
-    location: (tuple[str, str, str, str, int, int] | MessageLocationTuple),
+    location: tuple[str, str, str, str, int, int] | MessageLocationTuple,
```

Reply:
Added in follow-up!